server/datastore/mysql/vpp.go

```diff
@@ -779,6 +789,24 @@ func (ds *Datastore) UpdateVPPTokenTeams(ctx context.Context, id uint, teams []u
 			null_team_type
 	) VALUES `
 	stmtValues := `(?, ?, ?)`
+	// Delete all apps associated with a token if we change its team
+	stmtDeleteApps := `DELETE FROM vpp_apps_teams WHERE vpp_token_id = ?`
+	deleteArgs := []any{id}
+
+	if len(teams) > 0 {
+		// If we're adding a VPP token to one or more teams, delete
+		// any VPP apps already assigned to those teams (using the All
+		// teams token)
+		questions := make([]string, 0, len(teams))
+		for range len(teams) {
+			questions = append(questions, "?")
+		}
+		stmtDeleteApps += fmt.Sprintf(" OR team_id IN (%s)", strings.Join(questions, ","))
+
+		for _, team := range teams {
+			deleteArgs = append(deleteArgs, team)
+		}
+	}
```

Comment:
I was going to object this as all teams will lose all their apps if any other team assignment changes, but I think this is what you discussed with George, right?

Reply:
Yeah we discussed it, it's the fastest way to get the fix out right now without leaving any apps that potentially can't be used. We should revisit it soon but we need this to be done ASAP